### PR TITLE
Swap StakingPower and VotingPower, and modify from StakingPower to VotingWeight

### PR DESCRIPTION
**Pull request by tnasu**
## Description

Keeping tendermint's interface(abci, rpc, etc.) swap `StakingPower` and `VotingPower`.
And modify from `StakingPower` to `VotingWeight`

Here is the summary of the usage:
|    |  Tendermint  |  Ostracon (Before)  |  Ostracon (After)  |
| ---- | ---- | ---- | ---- |
|  Proposer  |  One of ValidatorSet  |  One of ValidatorSet  |  One of ValidatorSet  |
|  Select a Proposer by  |  [Validator.ProposerPriority](https://github.com/tendermint/tendermint/blob/v0.34.19/types/validator_set.go#L166)  |  [Validator.StakingPower](https://github.com/line/ostracon/blob/v1.0.6/types/validator_set.go#L653)  |  Validator.VotingPower  |
|  Voters  |  ValidatorSet  |  VoterSet (Part of ValidatorSet)   |  VoterSet (Part of ValidatorSet)  |
|  Select VoterSet by  |  -  |  [Validator.StakingPower](https://github.com/line/ostracon/blob/v1.0.6/types/voter_set.go#L488)  |  Validator.VotingPower  |
|  Voting  |  Validator.VotingPower  |  Validator.VotingPower  |  **_Validator.StakingPower_** => **_Validator.VotingWeight_**  |
|    |    |    |    |
|  DuplicateVoteEvidence TotalVotingPower  |  [ValidatorSet.TotalVotingPower](https://github.com/tendermint/tendermint/blob/v0.34.19/types/evidence.go#L69)  |  [VoterSet.TotalVotingPower](https://github.com/line/ostracon/blob/v1.0.6/types/evidence.go#L86)  |  **_VoterSet.TotalStakingPower_** => **_VoterSet.TotalVotingWeight_**  |
|  DuplicateVoteEvidence ValidatorPower  |  [Validator.VotingPower](https://github.com/tendermint/tendermint/blob/v0.34.19/types/evidence.go#L70)  |  [Validator.StakingPower](https://github.com/line/ostracon/blob/v1.0.6/types/evidence.go#L88)  |  Validator.VotingPower  |
|  DuplicateVoteEvidence added Power  |  -  |  [Validator.VotingPower](https://github.com/line/ostracon/blob/v1.0.6/types/evidence.go#L87)  |  **_Validator.StakingPower_** => **_Validator.VotingWeight_**  |
|  LightClientAttackEvidence TotalVotingPower  |  [ValidatorSet.TotalVotingPower](https://github.com/tendermint/tendermint/blob/v0.34.19/types/evidence.go#L196)  |  [VoterSet.TotalVotingPower](https://github.com/line/ostracon/blob/v1.0.6/types/evidence.go#L217)  |  **_VoterSet.TotalStakingPower_** => **_VoterSet.TotalVotingWeight_**  |
|    |    |    |    |
|  _ValidatorSet TotalXxxPower_  |  ValidatorSet.TotalVotingPower  |  VoterSet.TotalVotingPower  |  **_VoterSet.TotalStakingPower_** => **_VoterSet.TotalVotingWeight_**  |
|  Validator VotingPower  |  Validator.VotingPower  |  Validator.StakingPower  |  Validator.VotingPower  |
|  Validator added Power  |  -  |  Validator.VotingPower  |  **_Validator.StakingPower_** => **_Validator.VotingWeight_**  |
|  VoterSet TotalXxxPower  |  -  |  VoterSet.TotalVotingPower  |  **_VoterSet.TotalStakingPower_** => **_VoterSet.TotalVotingWeight_**  |
|    |    |    |    |
|  Proto Type DuplicateVoteEvidence TotalVotingPower  |  [ValidatorSet.TotalVotingPower](https://github.com/tendermint/tendermint/blob/v0.34.19/proto/tendermint/types/evidence.proto#L22)  |  [VoterSet.TotalVotingPower](https://github.com/line/ostracon/blob/v1.0.6/proto/ostracon/types/evidence.proto#L22)  |  **_VoterSet.TotalStakingPower_** => **_VoterSet.TotalVotingWeight_**  |
|  Proto Type DuplicateVoteEvidence Power  |  [Validator.VotingPower](https://github.com/tendermint/tendermint/blob/v0.34.19/proto/tendermint/types/evidence.proto#L23)  |  [Validator.StakingPower](https://github.com/line/ostracon/blob/v1.0.6/proto/ostracon/types/evidence.proto#L23)  |  Validator.VotingPower  |
|  Proto Type DuplicateVoteEvidence added Power  |  -  |  [Validator.VotingPower](https://github.com/line/ostracon/blob/v1.0.6/proto/ostracon/types/evidence.proto#L27)  |  **_Validator.StakingPower_** => **_Validator.VotingWeight_**  |
|  Proto Type LightClientAttackEvidence TotalVotingPower  |  [ValidatorSet.TotalVotingPower](https://github.com/tendermint/tendermint/blob/v0.34.19/proto/tendermint/types/evidence.proto#L32)  |  [VoterSet.TotalVotingPower](https://github.com/line/ostracon/blob/v1.0.6/proto/ostracon/types/evidence.proto#L35)  |  **_VoterSet.TotalStakingPower_** => **_VoterSet.TotalVotingWeight_**  |
|    |    |    |    |
|  _Proto Type ValidatorSet TotalXxxPower_  |  [ValidatorSet.TotalVotingPower](https://github.com/tendermint/tendermint/blob/v0.34.19/proto/tendermint/types/validator.proto#L12)  |  [VoterSet.TotalVotingPower](https://github.com/line/ostracon/blob/v1.0.6/proto/ostracon/types/validator.proto#L12)  |  **_VoterSet.TotalStakingPower_** => **_VoterSet.TotalVotingWeight_**  |
|  Proto Type Validator Power  |  [Validator.VotingPower](https://github.com/tendermint/tendermint/blob/v0.34.19/proto/tendermint/types/validator.proto#L18)  |  [Validator.StakingPower](https://github.com/line/ostracon/blob/v1.0.6/proto/ostracon/types/validator.proto#L18)  |  Validator.VotingPower  |
|  Proto Type Validator added Power  |  -  |  [Validator.VotingPower](https://github.com/line/ostracon/blob/v1.0.6/proto/ostracon/types/validator.proto#L22)  |  **_Validator.StakingPower_** => **_Validator.VotingWeight_**  |
|  Proto Type SimpleValidator Power  |  [Validator.VotingPower](https://github.com/tendermint/tendermint/blob/v0.34.19/proto/tendermint/types/validator.proto#L24)  |  [Validator.StakingPower](https://github.com/line/ostracon/blob/v1.0.6/proto/ostracon/types/validator.proto#L27)  |  Validator.VotingPower  |
|  _Proto Type VoterSet TotalXxxPower_  |  -  |  [VoterSet.TotalVotingPower](https://github.com/line/ostracon/blob/v1.0.6/proto/ostracon/types/voter.proto#L10)  |  **_VoterSet.TotalStakingPower_** => **_VoterSet.TotalVotingWeight_**  |
|    |    |    |    |
|  Proto ABCI Validator Power  |  Validator.VotingPower  |  Validator.StakingPower  |  Validator.VotingPower  |
|  Proto ABCI Validator added Power  |  -  |  Validator.VotingPower  |  **_Validator.StakingPower_** => **_Validator.VotingWeight_**  |
|  Proto ABCI ValidatorUpdate Power  |  Validator.VotingPower  |  Validator.StakingPower  |  Validator.VotingPower  |
|  Proto ABCI Evidence TotalVotingPower  |  ValidatorSet.TotalVotingPower  |  VoteSet.TotalVotingPower  |  **_VoterSet.TotalStakingPower_** => **_VoterSet.TotalVotingWeight_**  |
|    |    |    |    |
|  RPC ValidatorInfo VotingPower  |  [RPC ValidatorInfo VotingPower](https://github.com/tendermint/tendermint/blob/v0.34.19/rpc/core/types/responses.go#L91) == Validator.VotingPower  |  [RPC ValidatorInfo StakingPower](https://github.com/line/ostracon/blob/v1.0.6/rpc/core/types/responses.go#L91) == Validator.StakingPower  |  RPC ValidatorInfo VotingPower == Validator.VotingPower  |

* Tendermint.Validator.VotingPower  ==
  * Ostracon(Before).Validator.StakingPower == Ostracon(After).Validator.VotingPower
* Tendermint.Validator.VotingPower  !=
  * Ostracon(Before).Validator.VotingPower == Ostracon(After).Validator.StakingPower == **_Caluculated power value_**
* Tendermint.Evidence.TotalVotingPower  ==  Tendermint.ValidatorSet.TotalVotingPower !=
  * Ostracon(Before).VoterSet.TotalVotingPower == Ostracon(After).VoterSet.TotalStakingPower  == **_Caluculated power value_**

Relates:
* https://github.com/line/ostracon/pull/83
* https://github.com/line/ostracon/pull/102
* https://github.com/line/ostracon/pull/108
* https://github.com/line/ostracon/pull/110
* https://github.com/line/ostracon/pull/124
* https://github.com/line/ostracon/pull/129
* https://github.com/line/ostracon/pull/194 
* https://github.com/line/ostracon/pull/312
* https://github.com/line/lbm-sdk/pull/317

### CODE REVIEW

**Comment by tnasu** [comment]
I'd like to change the name of `Validator.StakingPower` since it's a bad name.

**Comment by tnasu** [comment]
> I'd like to change the name of Validator.StakingPower since it's a bad name.

I decide to use `VotingWeight` instead of `StakingPower`

**Comment by zemyblue** [comment]
* `VotingPower`: Voting power based on deposited tokens of validator.
* `VotingWeight`: Randomly allocated voting power when elected by VoterSet. 

@tnasu, Is it right?

**Comment by tnasu** [comment]
@zemyblue Right.

> Randomly allocated voting power when elected by VoterSet.

To explain it a little more carefully, the randomly selected `Validator` is the `VoterSet` from `ValidatorSet`, and when it is selected, the `VotingWeight` is calculated at the selection by using the `VotingPower` that the Validator has and the winning-point as a coefficient.

**Review by torao — DISMISSED** [review]
LGTM

File: abci/cmd/abci-cli/abci-cli.go [hunk 1]
```diff
@@ -55,8 +55,8 @@ var (
 	// kvstore
 	flagPersist string
 
-	// staking power for make validator_tx
-	flagStakingPower int64
+	// voting power for make validator_tx
+	flagVotingPower123456 int64
```
[the remark this answers is not in the document]
Comment (tnasu) [reply abci/cmd/abci-cli/abci-cli.go:59]: Oh, thanks, my temporary work is left... I'll fix it.

**Review by zemyblue — DISMISSED** [review]

**Review by torao — DISMISSED** [review]

File: types/voter_set_test.go [hunk 2]
```diff
@@ -845,15 +844,15 @@ func BenchmarkElectVotersNonDupEquity(b *testing.B) {
 	for i := 0; i < 99; i++ {
 		rewardRate := float64(accumulatedRewards[candidates.Validators[i].Address.String()]) /
 			float64(totalAccumulateRewards)
-		stakingRate := float64(candidates.Validators[i].StakingPower) / float64(totalStaking)
-		rate := rewardRate / stakingRate
-		rewardPerStakingDiff := math.Abs(1 - rate)
-		b.Log("rewardPerStakingDiff", rewardPerStakingDiff)
+		votingRate := float64(candidates.Validators[i].VotingPower) / float64(totalVoting)
+		rate := rewardRate / votingRate
+		rewardPerVotingWeightDiff := math.Abs(1 - rate)
```
Comment (tnasu) [inline types/voter_set_test.go:849]: rewardPerVotingPowerDiff?

File: types/voter_set_test.go [hunk 3]
```diff
@@ -867,16 +866,16 @@ func BenchmarkElectVotersNonDupEquity(b *testing.B) {
 		winners := electVotersNonDup(candidates.Validators, uint64(i), 20, 0)
 		accumulateAndResetReward(winners, accumulatedRewards)
 	}
-	maxRewardPerStakingDiff := float64(0)
+	maxRewardPerVotingWeightDiff := float64(0)
```
Comment (tnasu) [inline types/voter_set_test.go:869]: maxRewardPerVotingPowerDiff?

**Review by kokeshiM0chi — DISMISSED** [review]

**Review by kokeshiM0chi — APPROVED** [review]

**Review by torao — APPROVED** [review]

**Pull request merged**